### PR TITLE
chore: add assert for tp4 batch variant accuracy issue

### PR DESCRIPTION
**Pull request by yuki-97**
As title.

## Summary by CodeRabbit

* **Refactor**
  * Optimized loss function initialization workflow by consolidating setup operations.

* **Bug Fixes**
  * Added runtime validation safeguard for tensor model parallel configurations to ensure batch size consistency.

### CODE REVIEW

File: docs/guides/dtensor-tp-accuracy.md [hunk 1]
```diff
@@ -108,8 +108,6 @@ In batch-variant kernels, low-level implementation details—such as parallel re
 
 After aligning `train_micro_batch_size` and `logprob_batch_size` so that the same samples are processed with identical effective batch configurations, the importance-sampling ratio (`probs_ratio`) becomes 1 as expected, and the observed accuracy issues disappear. This confirms that the mismatch was caused by batch-dependent numerical variation rather than a conceptual error in the RL objective or data pipeline.
 
-Importantly, this problem is **model-specific** rather than universal. Models such as `deepseek-ai/DeepSeek-R1-Distill-Qwen-7B` and `Qwen/Qwen2.5-1.5B` exhibit clear batch-variant behavior under these settings, whereas `meta-llama/Llama-3.1-8B-Instruct` does not show the same sensitivity, likely due to differences in architecture, kernel implementations, or optimization choices in their respective stacks.
```
Comment (yuki-97) [inline docs/guides/dtensor-tp-accuracy.md:111]: I retested `meta-llama/Llama-3.1-8B-Instruct` TP4 with different train and logprobs batch size, it also has `ratio != 1`, so I removed this part.

File: examples/configs/recipes/llm/grpo-gemma3-27b-it-8n8g-fsdp2tp8-actckpt-long.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ policy:
   tokenizer:
     name: google/gemma-3-27b-it
   train_micro_batch_size: 1
-  logprob_batch_size: 2
```
Comment (terrykong) [inline examples/configs/recipes/llm/grpo-gemma3-27b-it-8n8g-fsdp2tp8-actckpt-long.yaml:14]: how about making the edit here:
https://github.com/NVIDIA-NeMo/RL/blob/main/examples/configs/grpo_math_1B.yaml#L78

and do something like:
```
logprob_batch_size: ${.train_micro_batch_size}
```

Comment (yuki-97) [reply examples/configs/recipes/llm/grpo-gemma3-27b-it-8n8g-fsdp2tp8-actckpt-long.yaml:14]: good idea! updated in https://github.com/NVIDIA-NeMo/RL/pull/1861/commits/bbc9d4b3ed7278f45e5195e1f70ce8a741fef6af, and also add an unit test to prevent in https://github.com/NVIDIA-NeMo/RL/pull/1861/changes/ea65045d4ac5b8d104f6456acefe2bfc6a7d293c.

Comment (terrykong) [reply examples/configs/recipes/llm/grpo-gemma3-27b-it-8n8g-fsdp2tp8-actckpt-long.yaml:14]: good idea on https://github.com/NVIDIA-NeMo/RL/commit/ea65045d4ac5b8d104f6456acefe2bfc6a7d293c.

**Review by terrykong — APPROVED** [review]
